lib/webapp.js

```diff
@@ -134,6 +149,51 @@ DataserviceContext.prototype = {
   makeErrorObject: zluxUtil.makeErrorObject
 };
 
+function createDataserviceStorage() {
+  return process.clusterManager.getStorage().then( function(clusterStorage) {
+    
+    clusterStorage.getStorage = function () {
+      return process.clusterManager.getStorage().then((storage) => {
+        clusterStorage.storageDict = storage.storageDict;
+        return storage.storageDict;
+      });
+    }
+
+    clusterStorage.setStorageValue = function (id, value) {
+      clusterStorage.storageDict[id] = value;
+      process.clusterManager.setStorageValue(id, value);
+    }
+
+    clusterStorage.setStorage = function (dict) {
+      clusterStorage.storageDict = dict;
+      process.clusterManager.setStorage(dict);
+    }
+
+    clusterStorage.addStorage = function (dict) {
+      clusterStorage.storageDict = { ...clusterStorage.storageDict, ...dict };
+      process.clusterManager.addStorage(dict);
+    }
+
+    clusterStorage.getStorageValue = function (id) {
+      process.clusterManager.getStorageValue(id).then((value) => {
+        return value;
+      });
+    }
+
+    clusterStorage.deleteStorageValue = function (id) {
+      delete clusterStorage.storageDict[id];
+      process.clusterManager.deleteStorageValue(id);
+    }
+
+    contentLogger.debug("App context is loaded with storage object: ", clusterStorage);
+    return clusterStorage;
+  })
+  .catch( function(e) { // Chances are, if storage failed to load, then the logger may have failed to load
```

Comment:
If you fail to get the cluster storage, then plugins that depend on this storage feature will not work as expected. It is time to stop the server.

Reply:
Plugins will still have storage, they just won't have the cluster storage. Still fatal?